src/rp2_common/pico_cyw43_driver/cyw43_driver.c

```diff
@@ -19,7 +19,23 @@
 #define CYW43_SLEEP_CHECK_MS 50
 #endif
 
-static async_context_t *cyw43_async_context;
+static async_context_t *cyw43_async_context = NULL;
+
+#if CYW43_USE_PARTITION_FIRMWARE
+    #include "pico/bootrom.h"
+    #include "hardware/flash.h"
+    #include "boot/picobin.h"
+    #include <stdlib.h>
+
+    // PICO_CONFIG: CYW43_WIFI_FW_PARTITION_ID, ID of Wi-Fi firmware partition, type=int, default=0x776966696669726d (wififirm), group=pico_cyw43_driver
+    #ifndef CYW43_WIFI_FW_PARTITION_ID
+    #define CYW43_WIFI_FW_PARTITION_ID 0x776966696669726d // wififirm
```

Comment:
I was humming an hah-ing about this, as i don't know whether we want

a) this to be generic (so that different WiFi chips can use the same ID)
b) non-specific with version info, so people can search for the right partition

I think for now this is OK (even though the code is cyw43_ specific) as we can always change it in the future

Reply:
I suspect that in the future we're likely to want to pass the `CYW43_WIFI_FW_PARTITION_ID` from the CMake in which case the `CYW43_USE_PARTITION_FIRMWARE` is perhaps superfluous, but i'm fine with it. (note i guess we won't pass ID from CMake for now as it would require dynamic JSON)

Please however rename `CYW43_WIFI_FW_PARTITION_ID` to `CYW43_WIFI_FIRMWARE_PARTITION_ID` as FW_ is not used as a contraction in any other defines... and I could go with `CYW43_USE_FIRMWARE_PARTITION`

Reply:
Finally, is this WiFi only; in which case either one or the other define should gain/lose `_WIFI`

Reply:
Will do - so CYW43_FIRMWARE_PARTITION_IF and CYW43_USE_FIRMWARE_PARTITION (I was just using WIFI_FW as CYW43_WIFI_FW_LEN is the define I was replacing from cyw43-driver)